docs/source/customizations/overview.mdx

```diff
@@ -20,36 +20,53 @@ If your plugin only needs to make basic changes to request or response headers,
 
 If not, the [hello world](https://github.com/apollographql/router/blob/225915f6e084b13c3a0972f170cf4a1a6f8e7f28/examples/hello-world/src/hello_world.rs) example plugin is a helpful starting point for writing your first [native Rust plugin](./native/).
 
-## Request lifecycle
+## How router plugins work
 
-All types of Apollo Router plugins hook into the router's core request lifecycle. It's helpful to understand the phases of this lifecycle and what data is available:
+Before building an Apollo Router plugin, it helps first to understand how the router handles each incoming GraphQL request. During each request's execution, four **services** in the router communicate with each other as shown:
 
 ```mermaid
 sequenceDiagram
-    participant Client
+    actor Client
```

Comment:
This makes Client a little stick person to differentiate it from the other items that are all part of the Router

Reply:
There are several other places where we can update this diagram. Will push later.